pkg/connector/tickets.go

```diff
@@ -0,0 +1,307 @@
+	}
+
+	var ret []*v2.TicketSchema
+	for _, team := range teams {
+		ret = append(ret, ticketSchemaFromTeam(team, customFields))
+	}
+
+	return ret, pageToken, annotations, nil
+}
+
+func ticketSchemaFromTeam(team linear.Team, customFields map[string]*v2.TicketCustomField) *v2.TicketSchema {
+	var statuses []*v2.TicketStatus
+
+	sort.Slice(team.States.Nodes, func(i, j int) bool {
+		return team.States.Nodes[i].Position < team.States.Nodes[j].Position
+	})
+
+	for _, state := range team.States.Nodes {
+		statuses = append(statuses, &v2.TicketStatus{
+			Id:          state.ID,
+			DisplayName: state.Name,
+		})
+	}
+
+	return &v2.TicketSchema{
+		Id:          team.ID,
+		DisplayName: team.Name,
+		Statuses:    statuses,
+		Types: []*v2.TicketType{
+			{
+				Id:          "issue",
+				DisplayName: "Issue",
+			},
+		},
+		CustomFields: customFields,
+	}
+}
+
+func (ln *Linear) getCustomFields(ctx context.Context) (map[string]*v2.TicketCustomField, error) {
+	fields, _, _, _, err := ln.client.ListIssueFields(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("baton-linear: failed to list issue fields: %w", err)
+	}
+
+	fieldMap := make(map[string]*v2.TicketCustomField)
+	for _, f := range fields {
+		if cfSchema, success := getCustomFieldSchema(f); success {
+			fieldMap[f.Name] = cfSchema
+		}
+	}
+
+	return fieldMap, nil
+}
+
+func getCustomFieldSchema(field linear.IssueField) (*v2.TicketCustomField, bool) {
+	if strings.HasPrefix(field.Description, "[Internal]") {
+		return nil, false
+	}
+	switch field.Name {
+	case "priority":
+		objectValues := []*v2.TicketCustomFieldObjectValue{
+			{Id: "0", DisplayName: "No priority"},
+			{Id: "1", DisplayName: "Urgent"},
+			{Id: "2", DisplayName: "High"},
+			{Id: "3", DisplayName: "Normal"},
+			{Id: "4", DisplayName: "Low"},
+		}
+		return sdkTicket.PickObjectValueFieldSchema(field.Name, field.Name, true, objectValues), true
+	case "assigneeId", "createAsUser", "cycleId", "projectId", "projectMilestoneId", "stateId", "subscriberIds", "templateId":
+		switch field.Type.Kind {
+		case "SCALAR":
+			switch field.Type.Name {
+			case "String":
+				return sdkTicket.StringFieldSchema(field.Name, field.Name, false), true
+			case "Boolean":
+				return sdkTicket.BoolFieldSchema(field.Name, field.Name, false), true
+			case "Float", "Int":
+				// At this time, NumberFieldSchema only supports Float and does not render properly in the UI.
+				// These need to be converted from String before being sent to Linear.
+				return sdkTicket.StringFieldSchema(field.Name, field.Name, false), true
+			case "JSON", "DateTime", "TimelessDate":
+				return nil, false
+			}
+		case "ENUM":
+			enums := make([]string, len(field.Type.EnumValues))
+			for i, v := range field.Type.EnumValues {
+				enums[i] = v.Name
+			}
+			return sdkTicket.PickMultipleStringsFieldSchema(field.Name, field.Name, false, enums), true
+		case "LIST":
+			return nil, false
```

Comment:
are none of these a list? (like `subscriberIds`?) or can that field not be set on create?